### PR TITLE
React intl markup

### PR DESCRIPTION
**Pull request by ddonahue007**
issue: https://issues.redhat.com/browse/COST-1372

updated createCostModelWizard/markup along with messages

Screen Shot 2021-08-31 at 2 25 47 PM

### CODE REVIEW

File: src/locales/messages.ts [hunk 1]
```diff
@@ -689,6 +694,26 @@ export default defineMessages({
...
+  CostModelsDetailsAssignSourceAriaLabel: {
+    defaultMessage: 'EN Assign source',
+    description: 'Assign source',
+    id: 'CostModelsDetailsAssignSourceAriaLabel',
+  },
+  CostModelsDetailsAssignSourcesLabel: {
+    defaultMessage: 'EN Assign source(s)',
+    description: 'Assign source(s)',
+    id: 'CostModelsDetailsAssignSources',
+  },
```
Comment (dlabrecq) [inline src/locales/messages.ts:711]: These strings are dups of CostModelsAssignSource and CostModelsAssignSources

File: src/locales/messages.ts [hunk 2]
```diff
@@ -803,6 +861,31 @@ export default defineMessages({
...
+  CostModelsWizardOnboardAWS: {
+    defaultMessage: 'EN Amazon Web Services (AWS)',
+    description: 'Amazon Web Services (AWS)',
+    id: 'CostModelsWizardOnboardAWS',
+  },
+  CostModelsWizardOnboardAzure: {
+    defaultMessage: 'EN Microsoft Azure',
+    description: 'Microsoft Azure',
+    id: 'CostModelsWizardOnboardAzure',
+  },
```
Comment (dlabrecq) [inline src/locales/messages.ts:883]: This is a dup of Azure

File: src/locales/messages.ts [hunk 3]
```diff
@@ -834,16 +917,73 @@ export default defineMessages({
     description: 'Assign sources to the cost model (optional)',
     id: 'CostModelsWizardSourceTitle',
   },
+  CostModelsWizardSourceTypeLabel: {
+    defaultMessage: 'EN Source type',
+    description: 'Source type',
+    id: 'CostModelsWizardSourceTypeLabel',
+  },
```
Comment (dlabrecq) [inline src/locales/messages.ts:924]: This is a dup of CostModelsSourceType

File: src/locales/messages.ts [hunk 4]
```diff
@@ -887,6 +1027,11 @@ export default defineMessages({
     description: 'total cost is the sum of the infrastructure cost and supplementary cost',
     id: 'DashboardTotalCostTooltip',
   },
+  Description: {
+    defaultMessage: 'EN description',
+    description: 'description',
+    id: 'Description',
+  },
```
Comment (dlabrecq) [inline src/locales/messages.ts:1034]: This is a dup of CostModelsTagRateTableDesc.

Thinking CostModelsTagRateTableDesc should be removed in favor of this one.

File: src/locales/messages.ts [hunk 5]
```diff
@@ -1028,13 +1173,37 @@ export default defineMessages({
...
+      'EN This choice is for users to direct how their raw costs are distributed either by CPU or Memory on the project level breakdowns.',
+    description:
+      'This choice is for users to direct how their raw costs are distributed either by CPU or Memory on the project level breakdowns.',
+    id: 'DistributionModelDesc',
+  },
+  DistributionType: {
+    defaultMessage: 'EN Distribution type',
+    description: 'Distribution type',
+    id: 'DistributionType',
+  },
```
Comment (dlabrecq) [inline src/locales/messages.ts:1192]: This is a dup of CostModelsDistributionType.

Thinking CostModelsDistributionType should be replaced with this one.

File: src/locales/messages.ts [hunk 6]
```diff
@@ -1698,18 +1889,18 @@ export default defineMessages({
     id: 'MonthOverMonthChange',
   },
   Name: {
-    defaultMessage: 'EN Name',
-    description: 'Name',
-    id: 'CostModelsWizardSourceTableName',
+    defaultMessage: '{count, plural, one {EN name} other {EN names}}',
+    description: 'Name plural or singular',
+    id: 'Name',
```
Comment (dlabrecq) [inline src/locales/messages.ts:1894]: We're using Name in a few places. 

If we're going to make this change, need to correct the other usages. It defaults to "names", which is not how it was intended to be used.

I see you changed two components below, but it's also used in:

dataToolbar.tsx (messages.Name)
addSourceWizard (messages.Name)
awsDetails/detailsTable (messages.Names)
addSourceStep (messages.Name)
explorerTable (messages.Names)
sourcesTable (messages.Name)
costModel/table (messages.Name)

File: src/locales/messages.ts [hunk 7]
```diff
@@ -1698,18 +1889,18 @@ export default defineMessages({
...
-    id: 'Names',
+  NameTitleCase: {
+    defaultMessage: '{count, plural, one {EN Name} other {EN Names}}',
+    description: 'Name in title case plural or singular',
+    id: 'NameTitleCase',
   },
   No: {
     defaultMessage: 'EN No',
-    description: 'No',
+    description: 'no',
```
Comment (dlabrecq) [inline src/locales/messages.ts:1903]: Lower case?

File: src/locales/messages.ts [hunk 8]
```diff
@@ -2071,6 +2262,11 @@ export default defineMessages({
     description: 'Perspective values',
     id: 'PerspectiveValues',
   },
+  Rate: {
+    defaultMessage: 'EN Rate',
+    description: 'Rate',
+    id: 'Rate',
+  },
```
Comment (dlabrecq) [inline src/locales/messages.ts:2269]: This is a dup of CostModelsTagRateTableRate

Thinking CostModelsTagRateTableRate should be replaced with this one.

**Review by dlabrecq — CHANGES_REQUESTED** [review]

File: src/locales/messages.ts (hunk 6 above)
Comment (dlabrecq) [reply src/locales/messages.ts:1894]: If the default is "names", then perhaps the message ID should also be Names?

File: src/locales/messages.ts (hunk 3 above)
Comment (ddonahue007) [reply src/locales/messages.ts:924]: Done

File: src/locales/messages.ts (hunk 6 above)
Comment (ddonahue007) [reply src/locales/messages.ts:1864]: OK for now until we discuss it I have changes those missed usages with a "count"

**Review by dlabrecq — APPROVED** [review]

**Pull request merged**